File: leetcode_problems/cpp/lc278_first_bad_version.cpp
bool isBadVersion(int version)
{
    return true;
}

class Solution
{
public:
    int firstBadVersion(int n)
    {
        int low = 0;
        int high = n;
        while (high - low > 1)
        {
            int mid = low + (high - low) / 2;
            if (isBadVersion(mid))
            {
                high = mid;
            }
            else
            {
                low = mid;
            }
        }
        return high;
    }
};

File: leetcode_problems/typescript/src/lc278_first_bad_version.ts
var solution = function (isBadVersion: any) {
  return function (n: number): number {
    let low = 0;
    let high = n;
    while (high - low > 1) {
      let mid = low + Math.floor((high - low) / 2);
      if (isBadVersion(mid)) {
        high = mid;
      } else {
        low = mid;
      }
    }
    return high;
  };
};
